addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1852,6 +1859,79 @@ def test_mail_composer_delete(self):
         self.assertEqual(len(self._new_msgs), 2, 'Should have created 1 mail.mail per record')
         self.assertFalse(self._new_msgs.exists(), 'Should have deleted mail.message records')
 
+    @users('employee')
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_mail_composer_duplicates(self):
+        """ Ensures emails sent to the same recipient multiple times
+            are only sent when they are not duplicates
+        """
+        self.template.write({
+            'auto_delete': False,
+            'body_html': '<p>Common Body</p>',
+            'email_to': 'test@test.lan',
+            'subject': 'Common Subject',
+            'partner_to': '',
+        })
+
+        # Guarantee points of variation for test_report_3
+        self.test_records[0].write({'name': 'A', 'count': 1})
+        self.test_records[1].write({'name': 'B', 'count': 2})
+
+        attachment = self.env['ir.attachment'].create({
+            'datas': base64.b64encode(b'ExtraData'),
+            'mimetype': 'text/plain',
+            'name': 'SomeCommonAttachment.txt',
+            'res_model': False,
+            'res_id': False,
+        })
+
+        def _instanciate_composer():
+            composer_form = Form(self.env['mail.compose.message'].with_context(
+                self._get_web_context(self.test_records, add_web=True,
+                                      default_template_id=self.template.id)
+            ))
+            return composer_form.save()
+
+        base = {
+            'attachment_ids': [Command.clear()],
+            'body_html': '<p>Common Body</p>',
+            'report_template_ids': [Command.clear()],
+            'subject': 'Common Subject',
+        }
+        same_attachments = {'attachment_ids': attachment.ids}
+        diff_body = {'body_html': '<p><t t-esc="object.name"></t></p>'}
+        diff_attachment_same_content = {'report_template_ids': [self.test_report_2.id]}
+        diff_attachment_diff_content = {'report_template_ids': [self.test_report_3.id]}
```

Comment:
Need the explanation from the previous comment to really understand the purpose here.

It looks like simply having "report_template_ids" will bypass the duplication check always, as "diff_attachment_same_content" is within the "don't detect duplicates case", yet we just compare lengths in the code, a bit lost.

Reply:
This check is currently not useful but I thought it was relevant to see that regardless of whether the templates are the same we currently assume they are not.

Globally I think it's good to have a test report with a variable body too, currently we only have different reports with the same body.

Will add a comment.